tests/loopbacked_tests.rs

```diff
@@ -33,19 +33,53 @@ use util::simple_tests::test_setup;
 use util::simple_tests::test_teardown;
 use util::simple_tests::test_thinpool_device;
 
+pub struct LoopTestDev {
+    ld: LoopDevice,
+}
+
+impl LoopTestDev {
+    pub fn new(lc: &LoopControl, path: &PathBuf) -> LoopTestDev {
+        OpenOptions::new()
+            .read(true)
+            .write(true)
+            .open(path)
+            .unwrap();
+
+        let ld = lc.next_free().unwrap();
+        ld.attach(path, 0).unwrap();
+        // Wipe 1 MiB at the beginning, as data sits around on the files.
+        wipe_sectors(&ld.get_path().unwrap(),
+                     Sectors(0),
+                     Bytes(IEC::Mi as u64).sectors())
+                .unwrap();
+
+        LoopTestDev { ld: ld }
+    }
+
+    fn get_path(&self) -> PathBuf {
+        self.ld.get_path().unwrap()
+    }
+
+    pub fn detach(&self) -> Result<()> {
+        self.ld.detach()
```

Comment:
To be consistant, you might as well unwrap() in the definition of the method, and then not unwrap in its use.